### ISSUE TITLE
Authorized followers: Easier way of removing individual accounts from followers list

### ISSUE THREAD

**Issue by bonsaiheldin** [tootsuite/mastodon#9398]
### Pitch

Hello there. At the moment, you can only remove all followers from an instance. If you want to remove an individual person from your followers list, you have to block and afterwards unblock them. My suggestion is to make this easier.

### Motivation

The "authorized followers" screen is not very helpful in this case as it only provides the ability to make whole instances unfollowing you (or not receiving your toots anymore) which is really impractical if you only want to remove a single person. A distinctive "Remove that account from your followers list"-button would help alot, similar to the features of blocking an account or a whole instance.

I would never have thought about blocking and unblocking until someone told me about that. I guess there are more people not knowing about it.

**Comment by Cassolotl** [comment]
This is a duplicate of *Ability to shove someone off your followers list (block-unblock shortcut) #8528*. :)

**Comment by bonsaiheldin** [comment]
@Cassolotl: Oh no! I even gave it a thumbs up. 😅 Seems i searched for the wrong words.

My issue can be closed. 😶

**Comment by Cassolotl** [comment]
@bonsaiheldin Easily done! I have done the same thing (including the thumbs-up part) many times! :P

**Comment by import-tiago** [comment]
Here a Python script to remove followers (by blocking them) who have more than 1k followers or are following more than 1k people. It reduced my followers by 50% (from approximately 120 to about 60). I hope this time I've gotten rid of these ghost followers.

```python
import requests

# Replace with your personal access token
TOKEN = "<personal-token>"
# Replace with your GitHub username
USERNAME = "<username>"

# Headers for the requests
headers = {
    "Authorization": f"token {TOKEN}",
    "Accept": "application/vnd.github+json"
}

# Step 1: Get your GitHub profile information (to print your follower count)
def get_my_profile():
    url = f"https://api.github.com/users/{USERNAME}"
    response = requests.get(url, headers=headers)
    if response.status_code == 200:
        return response.json()
    else:
        print(f"Failed to fetch your profile data. Status Code: {response.status_code}")
        return None

# Step 2: Get your list of followers with pagination support
def get_followers():
    followers = []
    page = 1
    while True:
        url = f"https://api.github.com/users/{USERNAME}/followers?per_page=100&page={page}"
        response = requests.get(url, headers=headers)
        if response.status_code == 200:
            data = response.json()
            if not data:  # If the data is empty, we've reached the last page
                break
            followers.extend(data)
            page += 1
        else:
            print(f"Failed to fetch followers. Status Code: {response.status_code}")
            break
    return followers

# Step 3: Get follower and following count for each follower
def get_follower_stats(follower_username):
    url = f"https://api.github.com/users/{follower_username}"
    response = requests.get(url, headers=headers)
    if response.status_code == 200:
        user_data = response.json()
        return user_data.get("followers", 0), user_data.get("following", 0)
    else:
        print(f"Failed to fetch data for {follower_username}. Status Code: {response.status_code}")
        return 0, 0

# Step 4: Block a user
def block_user(follower_username):
    url = f"https://api.github.com/user/blocks/{follower_username}"
    response = requests.put(url, headers=headers)
    if response.status_code == 204:
        print(f"Blocked {follower_username}")
    else:
        print(f"Failed to block {follower_username}. Status Code: {response.status_code}")

# Step 5: Check each follower and block if necessary
def block_followers_with_many_followers_or_following():
    followers = get_followers()
    print(f"Analyzing {len(followers)} followers.")
    
    for follower in followers:
        follower_username = follower["login"]
        follower_count, following_count = get_follower_stats(follower_username)
        
        if follower_count > 1000 or following_count > 1000:
            print(f"{follower_username} has {follower_count} followers and follows {following_count} users. Blocking...")
            block_user(follower_username)
        else:
            print(f"{follower_username} has {follower_count} followers and follows {following_count} users. No action taken.")

# Step 6: Main function to print your follower count and start the process
def main():
    profile = get_my_profile()
    if profile:
        my_follower_count = profile.get("followers", 0)
        print(f"You have {my_follower_count} followers.")
    
    # Proceed to check and block followers if necessary
    block_followers_with_many_followers_or_following()

# Execute the script
main()
```
